inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -283,6 +283,8 @@ public function save_post( $post_id ) {
 	 */
 	public function is_valid_post_type( $post_type ) {
 
+		global $wp_post_types;
```

Comment:
If some post type is registered _after_ this fires for the first time it would get invalid results from check. If performance wasn't a concern I would do an API call. If you take a look at `get_post_types()` source it's a very thin wrapper that just filters down global on each call.